### ISSUE TITLE
HasManyRepeater: watch for value change

### ISSUE THREAD

**Issue by zenati** [laravel-filament/filament#945]
### Package

filament/filament

### Package Version

2.5.31

### Laravel Version

8.76.2

### PHP Version

8.1

### Bug description

Inside a HasManyRepeater, I'm trying to show/hide a grid of fields, based on the value of a field. But the $get('type') is always empty.

### Steps to reproduce

I'm using the following function but it's not working.

```
->visible(function ($state, Closure $get, Model $model) {
      return $get('type') == 'DATE'; // <---- not working
  }),
```

<img width="680" alt="Capture d’écran 2021-12-20 à 12 11 50" src="https://user-images.githubusercontent.com/5849907/146758475-a045813d-3889-4e28-a97b-0f07456f13d5.png">

The idea is to show the grid of dates only if the type is DATE.

Here is the code so far:

```
Forms\Components\HasManyRepeater::make('fields')
    ->relationship('fields')
    ->schema([
        Forms\Components\Grid::make(8)->schema([
            Forms\Components\TextInput::make('label')->required()->columnSpan(5),
            Forms\Components\Select::make('type')->options(['STRING' => 'STRING', 'DATE' => 'DATE'])->reactive()->required()->columnSpan(3)
        ]),
        Forms\Components\Grid::make()->schema([
            Forms\Components\DatePicker::make('date_minimum'),
            Forms\Components\DatePicker::make('date_maximum')
        ])->visible(function ($state, Closure $get, Model $model) {
            return $get('type') == 'DATE'; // <---- not working
        }),
        Forms\Components\Grid::make(8)->schema([
            Forms\Components\Toggle::make('required')
        ])
])->label(false)
```

**Comment by danharrin** [comment]
Can you replace the FieldType stuff in the example please

**Comment by zenati** [comment]
@danharrin done.

**Comment by zenati** [comment]
@danharrin Do you need any other information regarding this issue ? Thanks.

**Comment by ryangjchandler** [comment]
@zenati I think we've got everything that we need. Please be patient as we investigate the problem.

**Comment by martinmildner** [comment]
hey, I have the same problem here:

```
HasManyRepeater::make('campaignElements')
    ->relationship('campaignElements')
    ->schema([
        BelongsToSelect::make('campaign_element_type_id')
            ->relationship('campaignElementType', 'title')
            ->reactive(),
         BelongsToSelect::make('calendar_id')
             ->relationship('calendar', 'title')
             ->visible(fn ($get) => $get('campaign_element_type_id') != 5),
```

$get('campaign_element_type_id') returns always null.

Resources ans relations:
- Campaign resource
- Pages relation manager
- Elements HasManyRepeater

**Comment by danharrin** [comment]
Fixed by #970.
